<a href="https://colab.research.google.com/github/AnnaVithoriaMS/Atividades_bootcamp_avanti/blob/main/Aula4_Estatistica_Descritiva.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Importação das bibliotecas

In [3]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
import warnings

In [ ]:
warnings.filterwarnings('ignore')

# Dados Pinguins

In [ ]:
pinguins = sns.load_dataset("penguins")

In [ ]:
pinguins.head()

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   

   body_mass_g     sex  
0       3750.0    Male  
1       3800.0  Female  
2       3250.0  Female  
3          NaN     NaN  
4       3450.0  Female

In [ ]:
pinguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     342 non-null    float64
 3   bill_depth_mm      342 non-null    float64
 4   flipper_length_mm  342 non-null    float64
 5   body_mass_g        342 non-null    float64
 6   sex                333 non-null    object 
dtypes: float64(4), object(3)
memory usage: 18.9+ KB


**Dados Quantitativos:**



- *bill_length_mm* :  comprimento do bico em milímetros

- *bill_depth_mm* :  altura do bico em milímetros
- *flipper_length_mm* : largura da asa em milímetros
- *body_mass_g* : massa corporal em gramas

**Dados Qualitativos**

**Nominal**

- species : uma das três espécies (Adélie, Chinstrap ou Gentoo)
- sex : sexo (female  ou male)
- island : uma das ilhas no arquipélago Palmer na Antartica (Biscoe, Dream ou Torgersen)

In [1]:
dic_dados_ping = {
    'bill_length_mm':['comprimento do bico','Quantitativos','Continuo'],
        'bill_depth_mm':['altura do bico','Quantitativos','Continuo'],
    'flipper_length_mm':[' largura da asa','Quantitativos','Continuo'],
    'body_mass_g':[' massa corporal','Quantitativos','Continuo'],
    'species': ['especie','Qualitativo','Nominal'],
    'sex': ['genero','Qualitativo','Nominal'],
    'island':['ilhas','Qualitativo','Nominal']
}

In [4]:
df_dic_ping = pd.DataFrame(dic_dados_ping)
df_dic_ping

bill_length_mm   bill_depth_mm flipper_length_mm      body_mass_g  \
0  comprimento do bico  altura do bico    largura da asa   massa corporal   
1        Quantitativos   Quantitativos     Quantitativos    Quantitativos   
2             Continuo        Continuo          Continuo         Continuo   

       species          sex       island  
0      especie       genero        ilhas  
1  Qualitativo  Qualitativo  Qualitativo  
2      Nominal      Nominal      Nominal

In [ ]:
pinguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

In [ ]:
pinguins.mode()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Biscoe            41.1           17.0              190.0   

   body_mass_g   sex  
0       3800.0  Male

In [ ]:
pinguins.skew()

bill_length_mm       0.053118
bill_depth_mm       -0.143465
flipper_length_mm    0.345682
body_mass_g          0.470329
dtype: float64

In [ ]:
q1 = pinguins.quantile(0.25)

q3 = pinguins.quantile(0.75)
iqr = q3 - q1
print(iqr)

bill_length_mm          9.275
bill_depth_mm           3.100
flipper_length_mm      23.000
body_mass_g          1200.000
dtype: float64


## **Comprimento do bico (bill_length_mm)**

- **MEDIA** :  43.92 mm

- **MEDIANA** : 44.45mm
- **MODA** : 41.1
- **DESVIO PADRAO** : 5.46
- **IQR** : 9.275
- **AMPLITUDE**:27.5

In [ ]:
max_comp = pinguins['bill_length_mm'].max()
min_comp  = pinguins['bill_length_mm'].min()

In [ ]:
dif_comp = max_comp -min_comp
dif_comp

27.5

##**Altura do bico  (bill_depth_mm)**

- **MEDIA** :  17.15 mm

- **MEDIANA** : 17.3mm
- **MODA** : 17.0 mm
- **DESVIO PADRAO** : 1.974
- **IQR** : 3.10
- **AMPLITUDE**: 8.4

In [ ]:
max_alt = pinguins['bill_depth_mm'].max()
min_alt  = pinguins['bill_depth_mm'].min()

In [ ]:
dif_alt = max_alt - min_alt
dif_alt

8.4

**Largura da asa (flipper_length_mm )**

**MEDIA** : 200.91mm

**MEDIANA** : 197mm

**MODA** : 190 mm

**DESVIO PADRAO** : 14.02

**IQR** : 23

**AMPLITUDE**: 59.0

In [ ]:
max_larg = pinguins['flipper_length_mm'].max()
min_larg  = pinguins['flipper_length_mm'].min()

In [ ]:
dif_larg = max_larg - min_larg
dif_larg

59.0

##**Massa corporal em gramas (body_mass_g)**

- **MEDIA** :  4201.75g

- **MEDIANA** : 4050g
- **MODA** : 3800g
- **DESVIO PADRAO** : 801.95
- **IQR** : 1200
- **AMPLITUDE**: 3600

In [ ]:
max_massa = pinguins['body_mass_g'].max()
min_massa  = pinguins['body_mass_g'].min()

In [ ]:
dif_massa = max_massa - min_massa
dif_massa

3600.0

##**Espécies**




In [ ]:

pinguins['species'].value_counts()

Adelie       152
Gentoo       124
Chinstrap     68
Name: species, dtype: int64

In [ ]:
pinguins['species'].value_counts(normalize = True)

Adelie       0.441860
Gentoo       0.360465
Chinstrap    0.197674
Name: species, dtype: float64

In [ ]:
ping_especie = pinguins.groupby('species')

### Media

In [ ]:
ping_especie.mean()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
species                                                                 
Adelie          38.791391      18.346358         189.953642  3700.662252
Chinstrap       48.833824      18.420588         195.823529  3733.088235
Gentoo          47.504878      14.982114         217.186992  5076.016260

### Mediana

In [ ]:
ping_especie.median()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
species                                                                 
Adelie              38.80          18.40              190.0       3700.0
Chinstrap           49.55          18.45              196.0       3700.0
Gentoo              47.30          15.00              216.0       5000.0

### Moda

In [ ]:
ping_especie.agg(lambda x:x.value_counts().index[0])

island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
species                                                               
Adelie      Dream            41.1           18.5              190.0   
Chinstrap   Dream            51.3           17.3              195.0   
Gentoo     Biscoe            46.5           15.0              215.0   

           body_mass_g     sex  
species                         
Adelie          3800.0    Male  
Chinstrap       3650.0  Female  
Gentoo          5000.0    Male

### Desvio Padrão

In [ ]:
import warnings
ping_especie.std()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
species                                                                 
Adelie           2.663405       1.216650           6.539457   458.566126
Chinstrap        3.339256       1.135395           7.131894   384.335081
Gentoo           3.081857       0.981220           6.484976   504.116237

### IQR

In [ ]:
q1 = ping_especie.quantile(0.25)

q3 = ping_especie.quantile(0.75)
iqr = q3 - q1
print(iqr)

           bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
species                                                                 
Adelie              4.000            1.5                9.0        650.0
Chinstrap           4.725            1.9               10.0        462.5
Gentoo              4.250            1.5                9.0        800.0


##**Ilhas**



In [ ]:
pinguins['island'].value_counts()

Biscoe       168
Dream        124
Torgersen     52
Name: island, dtype: int64

In [ ]:
pinguins['island'].value_counts(normalize = True)

Biscoe       0.488372
Dream        0.360465
Torgersen    0.151163
Name: island, dtype: float64

In [ ]:
ilhas = pinguins.groupby('island')

### Media

In [ ]:
ilhas.mean()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
island                                                                  
Biscoe          45.257485      15.874850         209.706587  4716.017964
Dream           44.167742      18.344355         193.072581  3712.903226
Torgersen       38.950980      18.429412         191.196078  3706.372549

### Mediana

In [ ]:
ilhas.median()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
island                                                                  
Biscoe              45.80           15.5              214.0       4775.0
Dream               44.65           18.4              193.0       3687.5
Torgersen           38.90           18.4              191.0       3700.0

### Moda

In [ ]:
ilhas.agg(lambda x:x.value_counts().index[0])

species  bill_length_mm  bill_depth_mm  flipper_length_mm  \
island                                                                   
Biscoe        Gentoo            45.2           15.0              215.0   
Dream      Chinstrap            36.0           18.5              190.0   
Torgersen     Adelie            34.6           17.6              190.0   

           body_mass_g     sex  
island                          
Biscoe          5000.0    Male  
Dream           3400.0    Male  
Torgersen       3700.0  Female

### Desvio Padrão

In [ ]:

ilhas.std()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
island                                                                  
Biscoe           4.772731       1.820721          14.142268   782.855743
Dream            5.953527       1.133116           7.507366   416.644112
Torgersen        3.025318       1.339447           6.232238   445.107940

### IQR

In [ ]:
q1 = ilhas.quantile(0.25)

q3 = ilhas.quantile(0.75)
iqr = q3 - q1
print(iqr)

           bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
island                                                                  
Biscoe               6.70            2.5              20.50      1125.00
Dream               10.70            1.5              10.25       556.25
Torgersen            4.45            1.9               8.00       662.50


##**Sexo**

In [ ]:
pinguins['sex'].value_counts()

Male      168
Female    165
Name: sex, dtype: int64

In [ ]:
pinguins['sex'].value_counts(normalize = True)

Male      0.504505
Female    0.495495
Name: sex, dtype: float64

In [ ]:
pinguins_sexo = pinguins.groupby('sex')

### Media

In [ ]:
pinguins_sexo.mean()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
sex                                                                  
Female       42.096970      16.425455         197.363636  3862.272727
Male         45.854762      17.891071         204.505952  4545.684524

### Mediana

In [ ]:
pinguins_sexo.median()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
sex                                                                  
Female            42.8          17.00              193.0       3650.0
Male              46.8          18.45              200.5       4300.0

### Moda

In [ ]:
pinguins_sexo.agg(lambda x:x.value_counts().index[0])

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
sex                                                                        
Female  Adelie  Biscoe            46.5           17.8              187.0   
Male    Adelie  Biscoe            41.1           18.5              197.0   

        body_mass_g  
sex                  
Female       3700.0  
Male         3900.0

### Desvio Padrão

In [ ]:
pinguins_sexo.std()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
sex                                                                  
Female        4.903476       1.795681          12.500776   666.172050
Male          5.366896       1.863351          14.547876   787.628884

### IQR

In [ ]:
q1 = pinguins_sexo.quantile(0.25)

q3 = pinguins_sexo.quantile(0.75)
iqr = q3 - q1
print(iqr)

        bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
sex                                                                  
Female            8.60          3.300               23.0       1200.0
Male              9.35          3.175               26.0       1412.5


# Dados Tips


In [ ]:
gorjeta = sns.load_dataset('tips')

In [ ]:
gorjeta.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [ ]:
gorjeta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype   
---  ------      --------------  -----   
 0   total_bill  244 non-null    float64 
 1   tip         244 non-null    float64 
 2   sex         244 non-null    category
 3   smoker      244 non-null    category
 4   day         244 non-null    category
 5   time        244 non-null    category
 6   size        244 non-null    int64   
dtypes: category(4), float64(2), int64(1)
memory usage: 7.4 KB


**Dados Quantitativos :**

- total_bill - Total da conta em dolar
- tips - Valor da gorjeta em dolar
- size - Numero de pessoas

**Dados qualitativos:**

- sex - Podendo ser Female (Feminino) ou Male (Masculino)
- smoker - Variavel pode ser Yes ou No
-day - Dias da Semana (quinta, sexta, sábado e domingo)
- time - Horario (Dinner para noite e Lunch para o dia)

São 244 amostras e 7 atributos

In [6]:
dic_dados_tips = {
    'total_bill':['Total da conta','Quantitativos','Continuo'],
        'tips':['Valor da gorjeta','Quantitativos','Continuo'],
    'size':['Numero de pessoas','Quantitativos','Discreto'],

    'smoker': ['fumantes','Qualitativo','Nominal'],
    'sex': ['genero','Qualitativo','Nominal'],
    'day':['Dias da Semana','Qualitativo','Nominal'],
    'time':['Horario','Qualitativo','Ordinal']
}

df_dados_tips = pd.DataFrame(dic_dados_tips)
df_dados_tips

total_bill              tips               size       smoker  \
0  Total da conta  Valor da gorjeta  Numero de pessoas     fumantes   
1   Quantitativos     Quantitativos      Quantitativos  Qualitativo   
2        Continuo          Continuo           Discreto      Nominal   

           sex             day         time  
0       genero  Dias da Semana      Horario  
1  Qualitativo     Qualitativo  Qualitativo  
2      Nominal         Nominal      Ordinal

In [ ]:
gorjeta.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

In [ ]:
gorjeta.mode()

total_bill  tip   sex smoker  day    time  size
0       13.42  2.0  Male     No  Sat  Dinner     2

In [ ]:
gorjeta.skew()

total_bill    1.133213
tip           1.465451
size          1.447882
dtype: float64

In [ ]:
q1 = gorjeta.quantile(0.25)

q3 = gorjeta.quantile(0.75)
iqr = q3 - q1
print(iqr)

total_bill    10.7800
tip            1.5625
size           1.0000
dtype: float64


##**Total - total_bill**

- **Media** 19.786
- **Mediana** 17.795
-  **Moda** 13.42
- **Desvio Padrão** 8.9
- **IQR** 10.78
- **Maximo** 50.81
- **Minimo** 3.07
- **Amplitude** 47.74

In [ ]:
max_total = gorjeta['total_bill'].max()
min_total = gorjeta['total_bill'].min()

dif_total = max_total - min_total
dif_total

47.74

##**Gorjeta - tips**
- **Media** 2.998
- **Mediana** 2.9
-  **Moda** 2.0
- **Desvio Padrão** 1.0
- **IQR** 1.56
-**Maximo** 1.0
-**Minimo** 10.0
- **Amplitude** 9.0

In [ ]:
max_tips = gorjeta['tip'].max()
min_tips = gorjeta['tip'].min()

dif_tips = max_tips - min_tips
dif_tips

9.0

##**Size**
- **Media** 2.57
- **Mediana** 2.0
-  **Moda** 2
- **Desvio Padrão** 0.95
- **IQR** 1.45
- **Maximo** 6
- **Minimo** 1
- **Amplitude** 5

In [ ]:
max_size = gorjeta['size'].max()
min_size= gorjeta['size'].min()

dif_size = max_size - min_size
dif_size

5

## **Sexo**



In [ ]:
gorjeta['sex'].value_counts()

Male      157
Female     87
Name: sex, dtype: int64

In [ ]:
gorjeta['sex'].value_counts(normalize = True)

Male      0.643443
Female    0.356557
Name: sex, dtype: float64

In [ ]:
sexo = gorjeta.groupby('sex')

### Media

In [ ]:
sexo.mean()

total_bill       tip      size
sex                                   
Male     20.744076  3.089618  2.630573
Female   18.056897  2.833448  2.459770

### Mediana

In [ ]:
sexo.median()

total_bill   tip  size
sex                           
Male         18.35  3.00   2.0
Female       16.40  2.75   2.0

### Moda

In [ ]:
sexo.agg(lambda x:x.value_counts().index[0])

total_bill  tip smoker   day    time  size
sex                                               
Male         10.34  2.0     No   Sat  Dinner     2
Female       10.33  2.0     No  Thur  Dinner     2

### Desvio Padrão

In [ ]:
sexo.std()

total_bill       tip      size
sex                                   
Male      9.246469  1.489102  0.955997
Female    8.009209  1.159495  0.937644

### IQR

In [ ]:
q1 = sexo.quantile(0.25)

q3 = sexo.quantile(0.75)
iqr = q3 - q1
print(iqr)

        total_bill   tip  size
sex                           
Male         10.71  1.76   1.0
Female        8.77  1.50   1.0


## **Smoker**



In [ ]:
gorjeta['smoker'].value_counts()

No     151
Yes     93
Name: smoker, dtype: int64

In [ ]:
gorjeta['smoker'].value_counts(normalize = True)

No     0.618852
Yes    0.381148
Name: smoker, dtype: float64

In [ ]:
smk= gorjeta.groupby('smoker')

### Media

In [ ]:
smk.mean()

total_bill       tip      size
smoker                                
Yes      20.756344  3.008710  2.408602
No       19.188278  2.991854  2.668874

### Mediana

In [ ]:
smk.median()

total_bill   tip  size
smoker                        
Yes          17.92  3.00   2.0
No           17.59  2.74   2.0

### Moda

In [ ]:
smk.agg(lambda x:x.value_counts().index[0])

total_bill  tip   sex  day    time  size
smoker                                          
Yes          13.42  2.0  Male  Sat  Dinner     2
No           15.98  2.0  Male  Sun  Dinner     2

### Desvio Padrão

In [ ]:
smk.std()

total_bill       tip      size
smoker                                
Yes       9.832154  1.401468  0.810751
No        8.255582  1.377190  1.017984

### IQR

In [ ]:
q1=smk.quantile(0.25)

q3 = smk.quantile(0.75)
iqr = q3 - q1
print(iqr)

        total_bill    tip  size
smoker                         
Yes          13.44  1.680   1.0
No            9.43  1.505   1.0


##**Dia - day**



In [ ]:
gorjeta['day'].value_counts()

Sat     87
Sun     76
Thur    62
Fri     19
Name: day, dtype: int64

In [ ]:
gorjeta['day'].value_counts(normalize = True)

Sat     0.356557
Sun     0.311475
Thur    0.254098
Fri     0.077869
Name: day, dtype: float64

In [ ]:
dia = gorjeta.groupby('day')

### Media

In [ ]:
dia.mean()

total_bill       tip      size
day                                 
Thur   17.682742  2.771452  2.451613
Fri    17.151579  2.734737  2.105263
Sat    20.441379  2.993103  2.517241
Sun    21.410000  3.255132  2.842105

### Mediana

In [ ]:
dia.median()

total_bill    tip  size
day                          
Thur       16.20  2.305   2.0
Fri        15.38  3.000   2.0
Sat        18.24  2.750   2.0
Sun        19.63  3.150   2.0

### Moda

In [ ]:
dia.agg(lambda x:x.value_counts().index[0])

total_bill  tip     sex smoker    time  size
day                                               
Thur       13.00  2.0  Female     No   Lunch     2
Fri        13.42  3.0    Male    Yes  Dinner     2
Sat        17.92  3.0    Male     No  Dinner     2
Sun        16.99  2.0    Male     No  Dinner     2

### Desvio Padrão

In [ ]:
dia.std()

total_bill       tip      size
day                                 
Thur    7.886170  1.240223  1.066285
Fri     8.302660  1.019577  0.567131
Sat     9.480419  1.631014  0.819275
Sun     8.832122  1.234880  1.007341

### IQR

In [ ]:
q1=dia.quantile(0.25)

q3 = dia.quantile(0.75)
iqr = q3 - q1
print(iqr)

      total_bill     tip  size
day                           
Thur      7.7125  1.3625   0.0
Fri       9.6550  1.4050   0.0
Sat      10.8350  1.3700   1.0
Sun      10.6100  1.9625   2.0


##**Time**



In [ ]:
gorjeta['time'].value_counts()

Dinner    176
Lunch      68
Name: time, dtype: int64

In [ ]:
gorjeta['time'].value_counts(normalize = True)

Dinner    0.721311
Lunch     0.278689
Name: time, dtype: float64

In [ ]:
hora = gorjeta.groupby('time')

### Media

In [ ]:
hora.mean()

total_bill       tip      size
time                                  
Lunch    17.168676  2.728088  2.411765
Dinner   20.797159  3.102670  2.630682

### Mediana

In [ ]:
hora.median()

total_bill   tip  size
time                          
Lunch       15.965  2.25   2.0
Dinner      18.390  3.00   2.0

### Moda

In [ ]:
hora.agg(lambda x:x.value_counts().index[0])

total_bill  tip     sex smoker   day  size
time                                              
Lunch        13.42  2.0  Female     No  Thur     2
Dinner       18.29  3.0    Male     No   Sat     2

### Desvio Padrão

In [ ]:
hora.std()

total_bill       tip      size
time                                  
Lunch     7.713882  1.205345  1.040024
Dinner    9.142029  1.436243  0.910241

### IQR

In [ ]:
q1 = hora.quantile(0.25)

q3 = hora.quantile(0.75)
iqr = q3 - q1
print(iqr)

        total_bill     tip  size
time                            
Lunch       7.2975  1.2875   0.0
Dinner     10.8450  1.6875   1.0


# Dados Titanic

In [ ]:
titanic = sns.load_dataset('titanic')

In [ ]:
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [ ]:
print("Tamanho: ", titanic.shape)
print("Nome dos Atributos: ", titanic.columns)

Tamanho:  (891, 15)
Nome dos Atributos:  Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')


In [ ]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


In [8]:
dic_dados_titanic = {
    'Age':['Idade','Quantitativos','Discreto'],
        'SibSp':[' Quantidade de cônjuges','Quantitativos','Discreto'],
    'Parch':['Quantidade de pais e filhos a bordo','Quantitativos','Discreto'],
    'Fare':['Preço da passagem','Quantitativos','Continuo'],
    'survived': [' Passageiros que sobreviveram','Qualitativo','Nominal'],
    'pclas': ['classe','Qualitativo','Ordinal'],
    'sex':['Genero','Qualitativo','Nominal'],
    'embarked':['Local de embarque','Qualitativo','Nominal'],
    'who':['Genero','Qualitativo','Nominal'],
    'Embark_town':['Local de embarque','Qualitativo','Nominal'],
    'alive':[' Passageiros que sobreviveram','Qualitativo','Nominal'],
    'alone':[' Passageiros desacompanhado','Qualitativo','Nominal']
}

df_dados_titanic = pd.DataFrame(dic_dados_titanic)
df_dados_titanic

Age                    SibSp  \
0          Idade   Quantidade de cônjuges   
1  Quantitativos            Quantitativos   
2       Discreto                 Discreto   

                                 Parch               Fare  \
0  Quantidade de pais e filhos a bordo  Preço da passagem   
1                        Quantitativos      Quantitativos   
2                             Discreto           Continuo   

                        survived        pclas          sex           embarked  \
0   Passageiros que sobreviveram       classe       Genero  Local de embarque   
1                    Qualitativo  Qualitativo  Qualitativo        Qualitativo   
2                        Nominal      Ordinal      Nominal            Nominal   

           who        Embark_town                          alive  \
0       Genero  Local de embarque   Passageiros que sobreviveram   
1  Qualitativo        Qualitativo                    Qualitativo   
2      Nominal            Nominal                        Nominal   

                         alone  
0   Passageiros desacompanhado  
1                  Qualitativo  
2                      Nominal

**Dados Quantitativos**

- **Age** - Idade do Passageiro
- **SibSp**- Quantidade de cônjuges e irmãos a bordo
- **Parch**- Quantidade de pais e filhos a bordo
- **Fare** - Preço da passagem

**Dados Qualitativos**
- **Survived** - Passageiros que sobreviveram
    - 0 = Não
    - 1 = Sim
- **Pclas** - Classe
    - 1 = Primeira classe
    - 2 = Segunda Classe
    - 3 = Terceira Classe
- **Sex** - Sexo do passageiro

    - Male - Masculino
    - Female - Feminino
- **Embarked** - Porto de emgaque
    - C - Cherbourg
    - Q - Queenstown
    - S - Southampton
- **Class** - Classe
    - First = Primeira classe
    - Second = Segunda Classe
    - Third = Terceira Classe
- **Who** - Genero
    - Man - Homem
    - Woman - Mulher
- **Embark_town** -  Cidade de embarque
    - Cherbourg
    - Queenstown
    - Southampton

- **Alive** - vivo
    - Yes - sim
    - No - não
- **Alone** - pessoas sozinhas
    - False
    - True

In [ ]:
titanic.describe()

survived      pclass         age       sibsp       parch        fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

In [ ]:
titanic.mode()

survived  pclass   sex   age  sibsp  parch  fare embarked  class  who  \
0         0       3  male  24.0      0      0  8.05        S  Third  man   

   adult_male deck  embark_town alive  alone  
0        True    C  Southampton    no   True

##**Idade - Age**

- Na coluna de idade há valores nulos, além de os valores aparecerem como float, não faz sentido uma idade 0.42.

- **Media** = 29.69 anos
- **Mediana** = 28 anos
- **Moda** = 24 anos
- **Desvio Padrão** = 14.52
- **IQR** = 17.875
- **Maximo** =  80
- **Minimo** = 0.42
- **Amplitude** = 79.58

In [ ]:
titanic['age'].unique()

array([22.  , 38.  , 26.  , 35.  ,   nan, 54.  ,  2.  , 27.  , 14.  ,
        4.  , 58.  , 20.  , 39.  , 55.  , 31.  , 34.  , 15.  , 28.  ,
        8.  , 19.  , 40.  , 66.  , 42.  , 21.  , 18.  ,  3.  ,  7.  ,
       49.  , 29.  , 65.  , 28.5 ,  5.  , 11.  , 45.  , 17.  , 32.  ,
       16.  , 25.  ,  0.83, 30.  , 33.  , 23.  , 24.  , 46.  , 59.  ,
       71.  , 37.  , 47.  , 14.5 , 70.5 , 32.5 , 12.  ,  9.  , 36.5 ,
       51.  , 55.5 , 40.5 , 44.  ,  1.  , 61.  , 56.  , 50.  , 36.  ,
       45.5 , 20.5 , 62.  , 41.  , 52.  , 63.  , 23.5 ,  0.92, 43.  ,
       60.  , 10.  , 64.  , 13.  , 48.  ,  0.75, 53.  , 57.  , 80.  ,
       70.  , 24.5 ,  6.  ,  0.67, 30.5 ,  0.42, 34.5 , 74.  ])

In [ ]:
q1 = titanic['age'].quantile(0.25)
q3 = titanic['age'].quantile(0.75)

In [ ]:
print('IQR')
iqr = q3 - q1
print(iqr)

IQR
17.875


In [ ]:
max_idade = titanic['age'].max()
min_idade = titanic['age'].min()

dif_idade = max_idade - min_idade
dif_idade

79.58

**Preço da passagem - Fare**


- **Media** = 32.2
- **Mediana** = 14.45
- **Moda** = 8.05
- **Desvio Padrão** = 49.7
- **IQR** = 23.09
- **Maximo** =  512.33
- **Minimo** = 0
- **Amplitude** = 512.33

In [ ]:
q1 = titanic['fare'].quantile(0.25)
q3 = titanic['fare'].quantile(0.75)
print('IQR')
iqr = q3 - q1
print(iqr)

IQR
23.0896


In [ ]:
max= titanic['fare'].max()
min= titanic['fare'].min()

dif = max - min
dif

512.3292

##**Classe - Pclass e Class**


In [ ]:
titanic['class'].value_counts()

Third     491
First     216
Second    184
Name: class, dtype: int64

In [ ]:
titanic['class'].value_counts(normalize = True)

Third     0.551066
First     0.242424
Second    0.206510
Name: class, dtype: float64

In [ ]:
classe = titanic.groupby('class')

### Media

In [ ]:
classe.mean()

survived  pclass        age     sibsp     parch       fare  \
class                                                                
First   0.629630     1.0  38.233441  0.416667  0.356481  84.154687   
Second  0.472826     2.0  29.877630  0.402174  0.380435  20.662183   
Third   0.242363     3.0  25.140620  0.615071  0.393075  13.675550   

        adult_male     alone  
class                         
First     0.550926  0.504630  
Second    0.538043  0.565217  
Third     0.649695  0.659878

### Mediana

In [ ]:
classe.median()

survived  pclass   age  sibsp  parch     fare  adult_male  alone
class                                                                   
First        1.0     1.0  37.0    0.0    0.0  60.2875         1.0    1.0
Second       0.0     2.0  29.0    0.0    0.0  14.2500         1.0    1.0
Third        0.0     3.0  24.0    0.0    0.0   8.0500         1.0    1.0

### Moda

In [ ]:
classe.agg(lambda x:x.value_counts().index[0])

survived  pclass   sex   age  sibsp  parch   fare embarked  who  \
class                                                                     
First          1       1  male  36.0      0      0  26.55        S  man   
Second         0       2  male  34.0      0      0  13.00        S  man   
Third          0       3  male  22.0      0      0   8.05        S  man   

        adult_male deck  embark_town alive  alone  
class                                              
First         True    C  Southampton   yes   True  
Second        True    F  Southampton    no   True  
Third         True    F  Southampton    no   True

### Desvio Padrão

In [ ]:
classe.std()

survived  pclass        age     sibsp     parch       fare  \
class                                                                
First   0.484026     0.0  14.802856  0.611898  0.693997  78.380373   
Second  0.500623     0.0  14.001077  0.601633  0.690963  13.417399   
Third   0.428949     0.0  12.495398  1.374883  0.888861  11.778142   

        adult_male     alone  
class                         
First     0.498555  0.501140  
Second    0.499911  0.497081  
Third     0.477552  0.474233

### IQR

In [ ]:
q1 = classe.quantile(0.25)

q3 = classe.quantile(0.75)
iqr = q3 - q1
print(iqr)

        survived  pclass   age  sibsp  parch      fare  adult_male  alone
class                                                                    
First        1.0     0.0  22.0    1.0    0.0  62.57605         1.0    1.0
Second       1.0     0.0  13.0    1.0    1.0  13.00000         1.0    1.0
Third        0.0     0.0  14.0    1.0    0.0   7.75000         1.0    1.0


##**Sexo - Sex e Who**


In [ ]:
titanic['sex'].value_counts()

male      577
female    314
Name: sex, dtype: int64

In [ ]:
titanic['sex'].value_counts(normalize = True)

male      0.647587
female    0.352413
Name: sex, dtype: float64

In [ ]:
sexo = titanic.groupby('sex')

### Media

In [ ]:
sexo.mean()

survived    pclass        age     sibsp     parch       fare  \
sex                                                                    
female  0.742038  2.159236  27.915709  0.694268  0.649682  44.479818   
male    0.188908  2.389948  30.726645  0.429809  0.235702  25.523893   

        adult_male     alone  
sex                           
female    0.000000  0.401274  
male      0.930676  0.712305

### Mediana

In [ ]:
sexo.median()

survived  pclass   age  sibsp  parch  fare  adult_male  alone
sex                                                                  
female       1.0     2.0  27.0    0.0    0.0  23.0         0.0    0.0
male         0.0     3.0  29.0    0.0    0.0  10.5         1.0    1.0

### Moda

In [ ]:
sexo.agg(lambda x:x.value_counts().index[0])

survived  pclass   age  sibsp  parch  fare embarked  class    who  \
sex                                                                         
female         1       3  24.0      0      0  7.75        S  Third  woman   
male           0       3  19.0      0      0  8.05        S  Third    man   

        adult_male deck  embark_town alive  alone  
sex                                                
female       False    B  Southampton   yes  False  
male          True    C  Southampton    no   True

### Desvio Padrão

In [ ]:
sexo.std()

survived   pclass        age     sibsp     parch       fare  \
sex                                                                   
female  0.438211  0.85729  14.110146  1.156520  1.022846  57.997698   
male    0.391775  0.81358  14.678201  1.061811  0.612294  43.138263   

        adult_male     alone  
sex                           
female    0.000000  0.490939  
male      0.254225  0.453081

### IQR

In [ ]:
q1 = sexo.quantile(0.25)

q3 = sexo.quantile(0.75)
iqr = q3 - q1
print(iqr)

        survived  pclass   age  sibsp  parch       fare  adult_male  alone
sex                                                                       
female       1.0     2.0  19.0    1.0    1.0  42.928125         0.0    1.0
male         0.0     1.0  18.0    0.0    0.0  18.654200         0.0    1.0


##**Sobreviventes - Survived e Alive**



In [ ]:
titanic['alive'].value_counts()

no     549
yes    342
Name: alive, dtype: int64

In [ ]:
titanic['alive'].value_counts(normalize = True)

no     0.616162
yes    0.383838
Name: alive, dtype: float64

In [ ]:
vivo = titanic.groupby('alive')

### Media

In [ ]:
vivo.mean()

survived    pclass        age     sibsp     parch       fare  \
alive                                                                 
no          0.0  2.531876  30.626179  0.553734  0.329690  22.117887   
yes         1.0  1.950292  28.343690  0.473684  0.464912  48.395408   

       adult_male     alone  
alive                        
no       0.817851  0.681239  
yes      0.257310  0.476608

### Mediana

In [ ]:
vivo.median()

survived  pclass   age  sibsp  parch  fare  adult_male  alone
alive                                                               
no          0.0     3.0  28.0    0.0    0.0  10.5         1.0    1.0
yes         1.0     2.0  28.0    0.0    0.0  26.0         0.0    0.0

### Moda

In [ ]:
vivo.agg(lambda x:x.value_counts().index[0])

survived  pclass     sex   age  sibsp  parch   fare embarked  class  \
alive                                                                        
no            0       3    male  21.0      0      0   8.05        S  Third   
yes           1       1  female  24.0      0      0  13.00        S  First   

         who  adult_male deck  embark_town  alone  
alive                                              
no       man        True    C  Southampton   True  
yes    woman       False    B  Southampton  False

### Desvio Padrão

In [ ]:
vivo.std()

survived    pclass        age     sibsp     parch       fare  \
alive                                                                 
no          0.0  0.735805  14.172110  1.288399  0.823166  31.388207   
yes         0.0  0.863321  14.950952  0.708688  0.771712  66.596998   

       adult_male     alone  
alive                        
no       0.386320  0.466421  
yes      0.437792  0.500184

### IQR

In [ ]:
q1 = vivo.quantile(0.25)

q3 = vivo.quantile(0.75)
iqr = q3 - q1
print(iqr)

       survived  pclass   age  sibsp  parch     fare  adult_male  alone
alive                                                                  
no          0.0     1.0  18.0    1.0    0.0  18.1458         0.0    1.0
yes         0.0     2.0  17.0    1.0    1.0  44.5250         1.0    1.0


##**Embarque  - Embark_town e Embarked**



In [ ]:
titanic['embark_town'].value_counts()

Southampton    644
Cherbourg      168
Queenstown      77
Name: embark_town, dtype: int64

In [ ]:
titanic['embark_town'].value_counts(normalize = True)

Southampton    0.724409
Cherbourg      0.188976
Queenstown     0.086614
Name: embark_town, dtype: float64

In [ ]:
embarque = titanic.groupby('embark_town')

### Media

In [ ]:
embarque.mean()

survived    pclass        age     sibsp     parch       fare  \
embark_town                                                                 
Cherbourg    0.553571  1.886905  30.814769  0.386905  0.363095  59.954144   
Queenstown   0.389610  2.909091  28.089286  0.428571  0.168831  13.276030   
Southampton  0.336957  2.350932  29.445397  0.571429  0.413043  27.079812   

             adult_male     alone  
embark_town                        
Cherbourg      0.535714  0.505952  
Queenstown     0.480519  0.740260  
Southampton    0.636646  0.610248

### Mediana

In [ ]:
embarque.median()

survived  pclass   age  sibsp  parch   fare  adult_male  alone
embark_town                                                                
Cherbourg         1.0     1.0  29.0    0.0    0.0  29.70         1.0    1.0
Queenstown        0.0     3.0  27.0    0.0    0.0   7.75         0.0    1.0
Southampton       0.0     3.0  28.0    0.0    0.0  13.00         1.0    1.0

### Moda

In [ ]:
embarque.agg(lambda x:x.value_counts().index[0])

survived  pclass   sex   age  sibsp  parch    fare embarked  \
embark_town                                                                
Cherbourg           1       1  male  30.0      0      0  7.2292        C   
Queenstown          0       3  male  21.0      0      0  7.7500        Q   
Southampton         0       3  male  28.0      0      0  8.0500        S   

             class  who  adult_male deck alive  alone  
embark_town                                            
Cherbourg    First  man        True    B   yes   True  
Queenstown   Third  man       False    C    no   True  
Southampton  Third  man        True    C    no   True

### Desvio Padrão

In [ ]:
embarque.std()

survived    pclass        age     sibsp     parch       fare  \
embark_town                                                                 
Cherbourg    0.498608  0.944100  15.434860  0.557213  0.660481  83.912994   
Queenstown   0.490860  0.369274  16.915396  0.979105  0.656999  14.188047   
Southampton  0.473037  0.789402  14.143192  1.216600  0.853253  35.887993   

             adult_male     alone  
embark_town                        
Cherbourg      0.500214  0.501459  
Queenstown     0.502897  0.441367  
Southampton    0.481339  0.488073

### IQR

In [ ]:
q1 = embarque.quantile(0.25)

q3 = embarque.quantile(0.75)
iqr = q3 - q1
print(iqr)

             survived  pclass    age  sibsp  parch       fare  adult_male  \
embark_town                                                                 
Cherbourg         1.0     2.0  18.75    1.0    1.0  64.802075         1.0   
Queenstown        1.0     0.0  17.00    0.0    0.0   7.750000         1.0   
Southampton       1.0     1.0  17.00    1.0    0.0  19.850000         1.0   

             alone  
embark_town         
Cherbourg      1.0  
Queenstown     1.0  
Southampton    1.0  


## **Sozinho - Alone**

In [ ]:
titanic['alone'].value_counts()

True     537
False    354
Name: alone, dtype: int64

In [ ]:
titanic['alone'].value_counts(normalize = True)

True     0.602694
False    0.397306
Name: alone, dtype: float64

In [ ]:
alone = titanic.groupby('alone')

### Media

In [ ]:
alone.mean()

survived    pclass        age     sibsp     parch       fare  \
alone                                                                 
False  0.505650  2.169492  26.413452  1.316384  0.960452  48.832275   
True   0.303538  2.400372  32.220297  0.000000  0.000000  21.242689   

       adult_male  
alone              
False    0.358757  
True     0.763501

### Mediana

In [ ]:
alone.median()

survived  pclass   age  sibsp  parch     fare  adult_male
alone                                                           
False       1.0     2.0  26.0    1.0    1.0  27.7500         0.0
True        0.0     3.0  29.5    0.0    0.0   8.1375         1.0

### Moda

In [ ]:
alone.agg(lambda x:x.value_counts().index[0])

survived  pclass     sex   age  sibsp  parch   fare embarked  class  \
alone                                                                        
False         1       3  female  24.0      1      0  26.00        S  Third   
True          0       3    male  22.0      0      0   8.05        S  Third   

         who  adult_male deck  embark_town alive  
alone                                             
False  woman       False    C  Southampton   yes  
True     man        True    B  Southampton    no

### Desvio Padrão

In [ ]:
alone.std()

survived    pclass        age     sibsp     parch       fare  \
alone                                                                 
False  0.500676  0.864520  15.834923  1.420774  1.039512  55.307615   
True   0.460214  0.804511  12.899871  0.000000  0.000000  42.223510   

       adult_male  
alone              
False    0.480315  
True     0.425328

### IQR

In [ ]:
q1 = alone.quantile(0.25)

q3 = alone.quantile(0.75)
iqr = q3 - q1
print(iqr)

       survived  pclass   age  sibsp  parch     fare  adult_male
alone                                                           
False       1.0     2.0  21.0    0.0    2.0  41.0448         1.0
True        1.0     1.0  17.0    0.0    0.0   7.2250         0.0


# Quarteto de Anscombe

In [ ]:
quarteto = sns.load_dataset("anscombe")

In [ ]:
quarteto.head()

dataset     x     y
0       I  10.0  8.04
1       I   8.0  6.95
2       I  13.0  7.58
3       I   9.0  8.81
4       I  11.0  8.33

In [ ]:
print("Tamanho: ", quarteto.shape)
print("Nome dos Atributos: ", quarteto.columns)

Tamanho:  (44, 3)
Nome dos Atributos:  Index(['dataset', 'x', 'y'], dtype='object')


**Dados Qualitativos**

* dataset
    * I, II, III, IV

**Dados Quantitativos**

    * x, y

In [10]:
dict_dado_quart = {
    'dataset':['Numero do dataset', 'Qualitativo','Nominal'],
    'x':['Ponto coordenada x', 'Quantitativo', 'Continuo'],
    'y':['Ponto coordenada y', 'Quantitativo', 'Continuo'],
}
df_quart = pd.DataFrame(dict_dado_quart)
df_quart

dataset                   x                   y
0  Numero do dataset  Ponto coordenada x  Ponto coordenada y
1        Qualitativo        Quantitativo        Quantitativo
2            Nominal            Continuo            Continuo

In [ ]:
quarteto.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44 entries, 0 to 43
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   dataset  44 non-null     object 
 1   x        44 non-null     float64
 2   y        44 non-null     float64
dtypes: float64(2), object(1)
memory usage: 1.2+ KB


In [ ]:
quarteto.describe()

x          y
count  44.000000  44.000000
mean    9.000000   7.500682
std     3.198837   1.958925
min     4.000000   3.100000
25%     7.000000   6.117500
50%     8.000000   7.520000
75%    11.000000   8.747500
max    19.000000  12.740000

In [ ]:
quarteto.mode()

dataset    x     y
0       I  8.0  8.84
1      II  NaN   NaN
2     III  NaN   NaN
3      IV  NaN   NaN

In [ ]:
quarteto.skew()

x    0.736875
y    0.439346
dtype: float64

In [ ]:
q1 = quarteto.quantile(0.25)
q3 = quarteto.quantile(0.75)
iqr = q3 - q1
print(iqr)

x    4.00
y    2.63
dtype: float64


In [ ]:
max = quarteto['x'].max()
min = quarteto['x'].min()

amp = max - min
print(amp)

15.0


In [ ]:
max = quarteto['y'].max()
min = quarteto['y'].min()

amp = max - min
print(amp)

9.64


## Dataset

In [ ]:
quarteto['dataset'].value_counts()

I      11
II     11
III    11
IV     11
Name: dataset, dtype: int64

In [ ]:
dados = quarteto.groupby('dataset')

In [ ]:
dados.describe()

x                                               y            \
        count mean       std  min  25%  50%   75%   max count      mean   
dataset                                                                   
I        11.0  9.0  3.316625  4.0  6.5  9.0  11.5  14.0  11.0  7.500909   
II       11.0  9.0  3.316625  4.0  6.5  9.0  11.5  14.0  11.0  7.500909   
III      11.0  9.0  3.316625  4.0  6.5  9.0  11.5  14.0  11.0  7.500000   
IV       11.0  9.0  3.316625  8.0  8.0  8.0   8.0  19.0  11.0  7.500909   

                                                   
              std   min    25%   50%   75%    max  
dataset                                            
I        2.031568  4.26  6.315  7.58  8.57  10.84  
II       2.031657  3.10  6.695  8.14  8.95   9.26  
III      2.030424  5.39  6.250  7.11  7.98  12.74  
IV       2.030579  5.25  6.170  7.04  8.19  12.50

In [ ]:
dados.agg(lambda x:x.value_counts().index[0])

x     y
dataset            
I        10.0  8.04
II       10.0  9.14
III      10.0  7.46
IV        8.0  6.58

In [ ]:
q1 = dados.quantile(0.25)

q3 = dados.quantile(0.75)
iqr = q3 - q1
print(iqr)

           x      y
dataset            
I        5.0  2.255
II       5.0  2.255
III      5.0  1.730
IV       0.0  2.020


In [ ]:
max = dados['x'].max()
min = dados['x'].min()

amp = max - min
print(amp)

dataset
I      10.0
II     10.0
III    10.0
IV     11.0
Name: x, dtype: float64


In [ ]:
max = dados['y'].max()
min = dados['y'].min()

amp = max - min
print(amp)

dataset
I      6.58
II     6.16
III    7.35
IV     7.25
Name: y, dtype: float64
